uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/toil-request.controller.js

```diff
@@ -57,43 +47,52 @@ define([
     function calculateToilExpiryDate () {
       // blocks the expiry date from updating if this is an existing request
       // and user is not a manager or admin
-      if (!vm.canManage && vm.request.id) {
-        return $q.resolve(vm.request.toil_expiry_date);
+      if (!parentCtrl.canManage && parentCtrl.request.id) {
```

Comment:
Looks to me that this should go in the `canCalculateExpiryDate` function

Reply:
@AkA84 Then canCalculateExpiryDate should be promise and this would mean the refactoring of all funcs. I would keep that in mind for the next time, but please let's do not increase the scope of this PR since it is already massive.

Reply:
Not sure what you mean, i just meant this
```js
function canCalculateExpiryDate () {
  var someConditionICantFigureOutForTheLifeOfMe = (!detailsController.canManage && detailsController.request.id);
  var multipleDaysWithToDateSet = (detailsController.uiOptions.multipleDays && detailsController.request.to_date);
  var singleDaysWithFromDateSet = (!detailsController.uiOptions.multipleDays && detailsController.request.from_date)

  return someConditionICantFigureOutForTheLifeOfMe && multipleDaysWithToDateSet || singleDaysWithFromDateSet;
}

function calculateToilExpiryDate () {
  // <-- condition is gone now
  if (!detailsController.requestCanExpire) {
    // ...
  }

  // rest of the function
```

Reply:
I see now. Thank you.

Reply:
Done.